sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/SpanDataAssert.java

```diff
@@ -159,6 +159,84 @@ public SpanDataAssert hasResource(Resource resource) {
     return this;
   }
 
+  /**
+   * Asserts the span has resource schemaUrl satisfying the given condition.
+   *
+   * @since 1.23.0
+   */
+  // Workaround "passing @Nullable parameter 'schemaUrl' where @NonNull is required", Nullaway
+  // seems to think assertThat is supposed to be passed NonNull even though we know that can't be
+  // true for assertions.
+  @SuppressWarnings("NullAway")
+  public SpanDataAssert hasResourceSchemaUrl(@Nullable String schemaUrl) {
+    isNotNull();
+    assertThat(actual.getResource().getSchemaUrl()).as("schemaUrl").isEqualTo(schemaUrl);
+    return this;
+  }
+
+  /**
+   * Asserts the span has resource attributes satisfying the given condition.
+   *
+   * @since 1.23.0
+   */
+  public SpanDataAssert hasResourceSatisfying(Consumer<Attributes> attributes) {
```

Comment:
Its weird to have a method named `hasResourceSatisfying` which accepts attributes.

How about we encapsulate these assertions in a new `ResourceAssert` class, and add a method with signature:
```
public SpanDataAssert hasResourceSatisfying(Consumer<ResourceAssert>);
```
Using would look like:
```
assertThat(span)
  .hasResourceSatisfying(resource -> resource
    .hasSchemaUrl("http://foo")
    .hasAttributes(..)
    .hasAttributesSatisfying(...));
```

Can also add `hasResourceSatisfying` to `MetricAssert`.

Reply:
I was trying to get away with the low hanging fruit approach, but surely you are right that its not consistent.

I'll need a bit time for the this, but I'm happy to make the change, also for the metrics.

Reply:
The PR is updated.